packages/zoe/src/typeGuards.js

```diff
@@ -78,3 +125,197 @@ export const InvitationElementShape = M.split({
+  getIssuerRecord: M.call().returns(IssuerRecordShape),
+  mintGains: M.call(AmountKeywordRecordShape, M.remotable()).returns(),
+  burnLosses: M.call(AmountKeywordRecordShape, M.remotable()).returns(),
+});
+
+export const ExitObjectShape = M.interface('Exit Object', {
+  exit: M.call().returns(),
+});
+
+export const InstanceAdminShape = M.interface('InstanceAdmin', {
+  makeInvitation: M.call(InvitationHandleShape, M.string())
+    .optional(M.any(), M.pattern())
+    .returns(M.promise()),
+  saveIssuer: M.call(M.eref(IssuerShape), M.string()).returns(M.promise()),
+  makeZoeMint: M.call(KeywordShape)
+    .optional(AssetKindShape, DisplayInfoShape, M.pattern())
+    .returns(M.remotable()),
+  registerFeeMint: M.call(KeywordShape, M.remotable()).returns(M.remotable()),
+  makeNoEscrowSeatKit: M.call(
+    AmountKeywordRecordShape,
+    ProposalShape,
+    M.remotable(),
+    SeatShape,
+  ).returns({ userSeat: SeatShape }),
+  replaceAllocations: M.call(SeatHandleAllocations).returns(),
+  exitAllSeats: M.call(M.any()).returns(),
+  failAllSeats: M.call(M.any()).returns(),
+  exitSeat: M.call(SeatShape, M.any()).returns(),
+  failSeat: M.call(SeatShape, M.any()).returns(),
+  stopAcceptingOffers: M.call().returns(),
+  setOfferFilter: M.call(M.arrayOf(M.string())).returns(),
+  getOfferFilter: M.call().returns(M.string()),
+  getExitSubscriber: M.call(SeatShape).returns(M.any()),
+});
+
+export const InstanceStorageManagerGuard = M.interface(
+  'InstanceStorageManager',
+  {
+    getTerms: M.call().returns(M.split(TermsShape, M.record())),
+    getIssuers: M.call().returns(IssuerKeywordRecordShape),
+    getBrands: M.call().returns(BrandKeywordRecordShape),
+    getInstallationForInstance: M.call().returns(InstallationShape),
+    getInvitationIssuer: M.call().returns(IssuerShape),
+
+    saveIssuer: M.call(IssuerShape, M.string()).returns(M.promise()),
+    makeZoeMint: M.call(KeywordShape)
+      .optional(AssetKindShape, DisplayInfoShape, M.pattern())
+      .returns(M.or(ZoeMintShape, M.remotable(), M.promise())),
+    registerFeeMint: M.call(KeywordShape, M.remotable()).returns(
+      IssuerKitShape,
+    ),
+    getInstanceRecord: M.call().returns(InstanceRecordShape),
+    getIssuerRecords: M.call().returns(M.arrayOf(IssuerRecordShape)),
+    getWithdrawPayments: M.call().returns(
+      M.interface('WithdrawFacet', {
+        withdrawPayments: M.call(AmountKeywordRecordShape).returns(
+          PaymentKeywordRecordShape,
+        ),
+      }),
+    ),
+    initInstanceAdmin: M.call(InstanceHandleShape, M.remotable()).returns(
+      M.promise(),
+    ),
+    deleteInstanceAdmin: M.call(InstanceAdminShape).returns(),
+    makeInvitation: M.call(InvitationHandleShape, M.string())
+      .optional(M.any(), M.pattern())
+      .returns(M.promise()),
+    getRoot: M.call().returns(M.promise()),
+    getAdminNode: M.call().returns(M.remotable()),
+  },
+);
+
+const BundleCapShape = M.any();
```

Comment:
In `SwingSet/src/types-external.js`, it is typed as `*`. I think sometimes it's a `remotable`, but other times, AFAICT, it might be a string.

I added a comment pointing to the source of my uncertainty.

Reply:
@kriskowal @warner  what kinds of shapes can BundleCaps have? Is there a more informative type declaration for these?

Reply:
@warner @kriskowal ping?

Reply:
It seems to be a device node. We're going to try treating it as an `M.remotable('bundleCap)`, but if that doesn't work, we'll back off to `M.any()`.

Reply:
I see this is done. Resolving.